### PR TITLE
Improve Fly mission example

### PR DESCRIPTION
**Pull request by shakthi-prashanth-m**
This is to improve Fly mission example based on issue #121.
Highlights:

- Adds [json11](https://github.com/dropbox/json11) as a submodule for parsing JSON files

- Example now handles waypoint missions fed from QGC plan (via command line arguments)

**Open question**: I feel the application code has more job to do. Can we provide an interface at Mission plugin ?

### CODE REVIEW

**Comment by shakthi-prashanth-m** [comment]
I strongly feel we can have add public method in Mission plugin which will do everything for the application.

File: example/fly_mission/fly_mission.cpp [hunk 1]
```diff
@@ -255,3 +253,99 @@ std::shared_ptr<MissionItem> add_mission_item(double latitude_deg,
...
+        auto command = mission_item["command"].int_value();
+
+        // Extract parameters
+        std::vector<double> params;
+        for (auto &p : mission_item["params"].array_items()) {
+            params.push_back(p.number_value());
+        }
+
+        // This looks bit complicated at app users.
+        // Should we handle them in plugin ?
```
Comment (julianoes) [inline example/fly_mission/fly_mission.cpp]: Yes, we should think of an import/export function inside the plugin.

**Comment by hamishwillee** [comment]
So just FYI this has been superseded by #235 , which adds the import/export function from a QGC plan within the module. 
I very much like the addition of the new API, but as documented in that PR, the example no longer shows much of the useful functionality. In either case, propose we close this item and proceed on the other PR?

**Comment by shakthi-prashanth-m** [comment]
@hamishwillee  yes. this PR mostly going to become obsolete when #235 is merged. However, until that PR is merged this one may be used by testing team (#233); hence this is kept open.

**Pull request closed without merging**